dot/peerset/peerset.go

```diff
@@ -402,7 +407,9 @@ func (ps *PeerSet) addReservedPeers(setID int, peers ...peer.ID) error {
 		}
 
 		ps.reservedNode[peerID] = struct{}{}
-		ps.peerState.addNoSlotNode(setID, peerID)
+		if err := ps.peerState.addNoSlotNode(setID, peerID); err != nil {
+			return err
```

Comment:
Feel free to add more context to the wrapping

```suggestion
			return fmt.Errorf("cannot add  no slot node: %w", err)
```

Reply:
addressed